src/util.cc

```diff
@@ -446,10 +446,11 @@ static void iterate_checksums(Task* t, ChecksumMode mode,
     FATAL() << "Failed to open checksum file " << filename;
   }
 
+  ReplaySession *replay = t->session().as_replay();
   remote_ptr<unsigned char> in_replay_flag;
   unsigned char in_replay = 0;
-  if (t->preload_globals) {
-    in_replay_flag = REMOTE_PTR_FIELD(t->preload_globals, in_replay);
+  if (replay && replay->has_trace_quirk(TraceReader::UsesGlobalsInReplay) && t->preload_globals) {
+    in_replay_flag = REMOTE_PTR_FIELD(t->preload_globals, reserved_legacy_in_replay);
     in_replay = t->read_mem(in_replay_flag);
```

Comment:
How can `in_replay` be anything other then 1 here?

Reply:
I suppose it can't anymore. I'll remove the memory read.